R/chat-tools.R

```diff
@@ -247,7 +247,52 @@ tool_results_as_turn <- function(results) {
   if (!any(is_tool_result)) {
     return(NULL)
   }
-  Turn("user", contents = results[is_tool_result])
+
+  results <- results[is_tool_result]
+
+  # Move any extra content (image/pdf) out of the tool result and into the user
+  # turn we use to encapsulate the tool results. This lets us support additional
+  # content types that aren't directly supported in tool results by most APIs.
+  extra_content <- list()
+  for (i in seq_along(results)) {
```

Comment:
This works, but it rewrites the tool result. Instead, I want to turn this part into a helper that separates tool results and the extra content in a way that can be called inside `as_json()`, i.e. just prior to any API call.